astropy/cosmology/tests/test_core.py

```diff
@@ -0,0 +1,555 @@
+        info = cosmo_cls.info
+        assert f"<astropy.cosmology.core.CosmologyInfo object" in repr(info)
+
+    # -------------------------------------------
+
+    def test_call_class_parameter(self, cosmo_cls, out, capsys):
+        """Test :meth:`astropy.cosmology.core.CosmologyInfo.__call__`."""
+        info = cosmo_cls.info(option="parameters", out=out)
+
+        if out is None:  # default
+            assert isinstance(info, MappingProxyType)
+            assert set(info.keys()) == set(cosmo_cls.__parameters__)
+            assert all((isinstance(v, Parameter) for v in info.values()))
+        else:  # various output options
+            if out == "":
+                r = capsys.readouterr().out
+            elif isinstance(out, StringIO):
+                r = out.getvalue()
+            elif isinstance(out, TextIOWrapper):
+                return  # TODO! figure out how to get output
+            else:
+                assert False, "SHOULD NEVER REACH THIS"
+
+            for p in cosmo_cls.__parameters__:
+                assert f"'{p}': <Parameter 'FlatLambdaCDM.{p}'" in r
+
+    def test_call_instance_parameter(self, cosmo, out, capsys):
+        """Test :meth:`astropy.cosmology.core.CosmologyInfo.__call__`."""
+        info = cosmo.info(option="parameters", out=out)
+
+        if out is None:  # default
+            assert isinstance(info, MappingProxyType)
+            assert set(info.keys()) == set(cosmo.__parameters__)
+
+        else:  # various output options
+            if out == "":
+                r = capsys.readouterr().out
+            elif isinstance(out, StringIO):
+                r = out.getvalue()
+            elif isinstance(out, TextIOWrapper):
+                return  # TODO! figure out how to get output
+            else:
+                assert False, "SHOULD NEVER REACH THIS"
+
+            for p in cosmo.__parameters__:
+                assert f"'{p}': " in r
+
+    def test_call_class_notoption(self, cosmo_cls, out):
+        """Test :meth:`astropy.cosmology.core.CosmologyInfo.__call__`."""
+        with pytest.raises(ValueError, match="'option' must be"):
+            cosmo_cls.info("not an option", out=out)
+
+    def test_call_instance_notoption(self, info, out):
+        """Test :meth:`astropy.cosmology.core.CosmologyInfo.__call__`."""
+        with pytest.raises(ValueError, match="'option' must be"):
+            info("not an option", out=out)
+
+
+# ========================================================================
+
+
+class TestParameter:
+    """Test `astropy.cosmology.Parameter`."""
+
+    def setup_class(self):
+        class Example1(Cosmology):
+            param = Parameter(unit=u.m, equivalencies=u.mass_energy(),
+                              doc="example parameter")
+
+            def __init__(self, param=15):
+                self._param = param
+
+            def __equiv__(self, other):  # needed to subclass Cosmology
```

Comment:
Maybe no abstract method! Also, indentation is off.

Reply:
Removed since ``__equiv__`` is no longer abstract.